feat: [Issue-185] Token Provider Tests

normand1 opened the pull request:
<!-- Use this template by filling in information and copy and pasting relevant items out of the html comments. -->

# Relates to:
https://github.com/ai16z/eliza/issues/185
<!-- LINK TO ISSUE OR TICKET -->

<!-- This risks section is to be filled out before final review and merge. -->

# Risks
Low: Only modification to src code files was adding optional syntax an

<!--
Low, medium, large. List what kind of risks, and what could be effected.
-->

# Background

## What does this PR do?
- Removes jest in favor of vitest as discussed in this issue: https://github.com/ai16z/eliza/issues/340
- Fixes tests that were no longer running in `packages/core/src/tests/token.test.ts`

## What kind of change is this?
Improvements - Tests
<!--
Bug fixes (non-breaking change which fixes an issue)
Improvements (misc. changes to existing features)
Features (non-breaking change which adds functionality)
Updates (new versions of included code)
-->

<!-- This "Why" section is most relevant if there is no linked issue explaining why. If there is a related issue it might make sense to skip this why section. -->
<!--
## Why are we doing this? Any context or related work?
-->
See Issue above

# Documentation changes needed?
My changes do not require a change to the project documentation.

<!--
My changes do not require a change to the project documentation.
My changes require a change to the project documentation.
If a docs change is needed: I have updated the documentation accordingly.
-->

<!-- Please show how you tested the PR. This will really help if the PR needs to be retested, and probably help the PR get merged quicker. -->

# Testing
`pnpm -w run test -- 'src/tests/token.test.ts'`

## Where should a reviewer start?
```
pnpm install
pnpm run build
pnpm -w run test -- 'src/tests/token.test.ts'
```

## Detailed testing steps

<!--
None, automtated tests are fine.
-->

<!--
- As [anon/admin], go to [link]
  - [do action]
  - verify [result]
-->

<!-- If there is a UI change, please include before and after screenshots or videos. This will speed up PRs being merged. It is extra nice to annotate screenshots with arrows or boxes pointing out the differences. -->
<!--
## Screenshots
### Before
### After
-->

<!-- If there is anything about the deploy, please make a note. -->
<!--
# Deploy Notes
-->

<!--  Copy and paste commandline output. -->
<!--
## Database changes
-->

<!--  If there is something more than the automated steps, please specifiy deploy instructions. -->
<!--
## Deployment instructions
-->

normand1 commented:
@ponderingdemocritus This is based on your earlier instructions to include and use vitest:
https://github.com/ai16z/eliza/issues/340#issuecomment-2480255525

ponderingdemocritus commented on packages/adapter-sqlite/package.json:
```diff
@@ -19,5 +19,9 @@
+    },
+    "exports": {
+        ".": "./dist/index.js",
+        "./sqlite_vec": "./dist/sqlite_vec.js"
```
i don't think this is needed - exported at index

ponderingdemocritus commented on packages/plugin-solana/src/index.ts:
```diff
@@ -12,6 +12,12 @@ import { Plugin } from "@ai16z/eliza";
+import { Connection } from "@solana/web3.js";
+import { PublicKey } from "@solana/web3.js";
+
+export { TokenProvider, WalletProvider, Connection, PublicKey };
```
remove the solana exports and import where needed from the packages

normand1 replied:
Good catch, this was left over from another test I was working on, but ended up simplifying for this first PR

ponderingdemocritus reviewed (commented):
good start - see comments

normand1 commented:
@ponderingdemocritus Thanks for the feedback! Just pushed an update based on feedback

monilpat commented on packages/core/src/tests/token.test.ts:
```diff
@@ -0,0 +1,76 @@
+        
+        // Create new instance of TokenProvider
+        tokenProvider = new TokenProvider(
+            "2weMjPLLybRMMva1fM3U31goWWrCpF59CHWNhnCJ9Vyh",
+            new WalletProvider(
+                new Connection("https://api.mainnet-beta.solana.com"),
+                new PublicKey("2weMjPLLybRMMva1fM3U31goWWrCpF59CHWNhnCJ9Vyh")
+            )
```
Instead of hard coding these can we move these into a constants file?

monilpat commented on packages/core/src/tests/token.test.ts:
```diff
@@ -0,0 +1,76 @@
+            top10HolderPercent: 20,
+        });
+
+        // Ensure the mock was called
+        expect(fetchSpy).toHaveBeenCalled();
+
```
We should remove this as the line below does the same check and there this is redundant

monilpat commented on packages/core/src/test_resources/types.ts:
```diff
@@ -0,0 +1,6 @@
+export interface User {
```
I am surprised that this type hasn't already been declared and can't be reused. Could be mistaken but wanted to point it out thanks

normand1 replied:
This type is intended to generically represent the different User types used across packages in the agent runtime based on whether we're using sqlite, supabase or sqljs. I couldn't find an abstract type that currently exists for this, but I'd be happy to replace it here if there is one. If there isn't one then I agree this would be helpful to have a real type that does this instead of this one sitting in a test file :)

monilpat commented on packages/core/src/test_resources/createRuntime.ts:
```diff
@@ -0,0 +1,146 @@
+                // Load sqlite-vss
+                loadVecExtensions((adapter as SqlJsDatabaseAdapter).db);
+                // Create a test user and session
+                user = {
+                    id: zeroUuid,
+                    email: "test@example.com",
+                } as User;
+                session = {
+                    user: user,
+                };
```
Nit: user can be inlined here

monilpat commented on packages/core/src/test_resources/createRuntime.ts:
```diff
@@ -0,0 +1,146 @@
+                // Load sqlite-vss
+                await loadVecExtensions((adapter as SqliteDatabaseAdapter).db);
+                // Create a test user and session
+                user = {
+                    id: zeroUuid,
+                    email: "test@example.com",
+                } as User;
+                session = {
+                    user: user,
+                };
```
Same here

monilpat commented on packages/core/src/test_resources/createRuntime.ts:
```diff
@@ -0,0 +1,146 @@
+    }
+
+    const runtime = new AgentRuntime({
+        serverUrl: "https://api.openai.com/v1",
```
probably a const we can use here

monilpat commented on packages/core/src/test_resources/basic.test.ts:
```diff
@@ -0,0 +1,12 @@
+import { describe, expect, it } from "vitest";
+
+describe("Basic Test Suite", () => {
+    it("should run a basic test", () => {
```
Please remove this

monilpat reviewed (changes requested):
Overall looks good thanks for doing this - some code quality changes to make then can accept!

normand1 commented:
> Thanks for working on this. It's ready to go after addressing comments

Awesome thank you for the review, taking a look at making those updates now

monilpat commented:
Also, forgot to mention please address the merge conflicts thanks :)

normand1 commented:
@monilpat I believe I've addressed all feedback, ready for someone to take another look, thanks!

normand1 commented:
@jkbrooks Resolved conflicts

monilpat commented:
Cool with the resolved conflicts LGTM!
